testing/src/test/java/org/motechproject/nms/testing/it/kilkari/MctsBeneficiaryUpdateServiceBundleIT.java

```diff
@@ -0,0 +1,525 @@
+        Subscriber oldSubscriber2 = subscriberDataService.findByCallingNumber(mother2msisdn);
+        assertNull(oldSubscriber2.getMother());
+        assertEquals(0, oldSubscriber2.getAllSubscriptions().size());
+
+        Subscriber subscriber2 = subscriberDataService.findByCallingNumber(9439986188L);
+        assertNotNull(subscriber2);
+        assertEquals(mother2id, subscriber2.getMother().getBeneficiaryId());
+
+        // Location update:
+        MctsMother updatedMother3 = mctsMotherDataService.findByBeneficiaryId(mother3id);
+        assertEquals(21L, (long) updatedMother3.getState().getCode());
+        assertEquals(3L, (long) updatedMother3.getDistrict().getCode());
+        assertEquals("0026", updatedMother3.getTaluka().getCode());
+        assertEquals(453L, (long) updatedMother3.getHealthBlock().getCode());
+
+        // DOB update:
+        String updatedDOB = "01-07-2015";
+        Subscriber subscriber4 = subscriberDataService.findByCallingNumber(child4msisdn);
+        assertEquals(updatedDOB, getDateString(subscriber4.getDateOfBirth()));
+        Subscription updatedSubscription = subscriber4.getActiveAndPendingSubscriptions().iterator().next();
+        assertEquals(updatedDOB, getDateString(updatedSubscription.getStartDate()));
+        assertEquals(SubscriptionStatus.ACTIVE, updatedSubscription.getStatus());
+    }
+
+    private void makeMctsSubscription(MctsBeneficiary beneficiary, DateTime startDate, SubscriptionPackType packType, Long number) {
+        subscriptionHelper.mksub(SubscriptionOrigin.MCTS_IMPORT, startDate, packType, number);
+        Subscriber subscriber = subscriberDataService.findByCallingNumber(number);
+        if (packType == SubscriptionPackType.CHILD) {
+            subscriber.setChild((MctsChild) beneficiary);
+            subscriber.setDateOfBirth(startDate);
+        } else {
+            subscriber.setMother((MctsMother) beneficiary);
+            subscriber.setLastMenstrualPeriod(startDate.minusDays(90));
+        }
+        subscriberDataService.update(subscriber);
+    }
+
+    private Reader createUpdateReaderWithHeaders(String... lines) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Sr No,MCTS ID,STATE ID,Beneficiary New DOB change,Beneficiary New LMP change,StateID,"); // 6 columns
+        builder.append("District_ID,Taluka_ID,HealthBlock_ID,PHC_ID,SubCentre_ID,Village_ID,GP_Village,Address,"); // +8 columns
+        builder.append("Beneficiary New Mobile no change"); // +1 column
+        builder.append("\n");
+
+        for (String line : lines) {
+            builder.append(line).append("\n");
+        }
+
+        return new StringReader(builder.toString());
+    }
+
+    private Reader read(String resource) {
+        return new InputStreamReader(getClass().getClassLoader().getResourceAsStream(resource));
+    }
+
+    private String getDateString(DateTime date) {
+        return date.toString("dd-MM-yyyy");
+    }
+
+}
```

Comment:
How about testing some failure cases and making sure you're getting the right errors/exceptions/alerts?